hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCompleteRequest.java

```diff
@@ -229,6 +229,9 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
               trxnLogIndex, ozoneManager.isRatisEnabled());
           long numCopy = keyToDelete.getReplicationConfig().getRequiredNodes();
           usedBytesDiff -= keyToDelete.getDataSize() * numCopy;
+        } else {
+          checkBucketQuotaInNamespace(omBucketInfo, 1L);
+          omBucketInfo.incrUsedNamespace(1L);
```

Comment:
The update in `omBucketInfo` is lost if this is not an overwrite (or an overwrite with no size diff):

https://github.com/apache/ozone/blob/86ffd947350ede008893283636c86cb1b5d31b38/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCompleteRequest.java#L240-L241